### ISSUE TITLE
The headerRight/headerLeft buttons does not work on Android

### ISSUE THREAD

**Issue by sayanasay** [react-navigation/react-navigation#12274]
### Current behavior

The button in the header (headerLeft and headerRight) does not work on Android. The button visually responds to the touch (I use TouchableOpacity), but the action is not executed. This issue appeared after updating to Expo SDK 52. It works correctly on iOS.

### Expected behavior

The button should respond to the touch and execute the action.

### Reproduction

https://snack.expo.dev/@sayanasay/react-navigation-headerleft-button

### Platform

- [x] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [x] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |^7.0.3
| @react-navigation/bottom-tabs          |
| @react-navigation/drawer               |
| @react-navigation/material-top-tabs    |
| @react-navigation/stack                |
| @react-navigation/native-stack         |^7.0.4
| react-native-safe-area-context         |4.12.0
| react-native-screens                   |~4.0.0
| react-native-gesture-handler           |
| react-native-reanimated                |
| react-native-tab-view                  |
| react-native-pager-view                |
| react-native                           |0.76.2
| expo                                   |~52.0.7
| node                                   |
| npm or yarn                            |

**Comment by LegendProjects** [comment]
> is this because of [software-mansion/react-native-screens#2219](https://github.com/software-mansion/react-native-screens/issues/2219) ?

No, currently in this the headerStyle work for both Large and Small header and I cannot change color like before to the Large header and I set headerLargeStyle because I use transparent and blureffect but before worked normal and now doesn't work at all, so I cannot make like iOS the settings menu blur when is small and no blur static solid black when is large.

**Comment by gigilibala** [comment]
I have the same issue. Although only on a release android build on actual phone. Android emulator works fine. Here is an example app that reproduces the issue.

https://github.com/gigilibala/testscreenbug

Please, fix. 
Thank you.

**Comment by TedKoDev** [comment]
https://github.com/software-mansion/react-native-screens/issues/2219#issuecomment-2481628312

it is work not but it is temperary.

**Comment by kriit24** [comment]
same thing in ios [https://github.com/expo/expo/issues/32927](https://github.com/expo/expo/issues/32927)

**Comment by serdarovez** [comment]
can someone help me too if i put image on headerTitle/headerLeft/headerRight the header takes entire space of screen on android after update how can i fix it?

**Comment by Hakanlsk** [comment]
I was having the same problem and this is how I solved it for now:

`"newArchEnabled":false,`

**Comment by jaseembinmuhammed** [comment]
Hei there 
Use onPressOut instead of onPress 
Will work

**Comment by serdarovez** [comment]
> Hei there Use onPressOut instead of onPress Will work

not working :(

**Comment by kriit24** [comment]
onPressIn worked for me, but this is not the solution, this is workaround.
solution is to fix ASAP

**Comment by metacop** [comment]
Using RectButton from react-native-gesture-handler is a workaround in this case.

example:

```

headerLeft: ({ canGoBack, tintColor }) => (
          <RectButton onPress={() => onHeaderLeftPress(canGoBack)}>
            <Ionicons
              name={canGoBack ? 'arrow-back-outline' : 'menu-outline'}
              size={24}
              onPress={() => {
                console.log({ canGoBack });
                onHeaderLeftPress(canGoBack);
              }}
              className="mr-5"
            />
          </RectButton>
        ),

```

also if they use react navigate 7, the goback must be per router.

```
const onHeaderLeftPress = (canGoBack: boolean = false) => {
    if (canGoBack) {
      router.back();
      return;
    }
    navigation.dispatch(DrawerActions.toggleDrawer());
  };
```

**Comment by LasithaOffice** [comment]
> Hei there Use onPressOut instead of onPress Will work

This worked for me. Thank you

**Comment by Reqima-dev** [comment]
> Hei there Use onPressOut instead of onPress Will work

This worked for me. Thank you

**Comment by MobileMon** [comment]
this is still an issue

onPressOut is not a good solution because if the button is in a viewPager and you scroll while its highlighted, it will trigger the button press when its not supposed to 

this bug needs to be fixed

**Comment by markpw7** [comment]
I use HeaderBackButton from '@react-navigation/elements' which does not accept an onPressOut prop. 

Please fix this issue.

**Comment by oluwajuwonemmanuel** [comment]
> Hei there As a workaround, use onPressOut instead of onPress Will work

Wow thank you! it worked

**Comment by oluwajuwonemmanuel** [comment]
> this is still an issue
> 
> onPressOut is not a good solution because if the button is in a viewPager and you scroll while its highlighted, it will trigger the button press when its not supposed to
> 
> this bug needs to be fixed

Please do you have another solution?

**Comment by ItaloFalzoni** [comment]
Try to use TouchableOpacity of react-native-gesture-handler
```
import { TouchableOpacity } from "react-native-gesture-handler";

...

<TouchableOpacity onPress={() => {}} />
```

**Comment by MobileMon-Majestic** [comment]
> Try to use TouchableOpacity of react-native-gesture-handler
> 
> ```
> import { TouchableOpacity } from "react-native-gesture-handler";
> 
> ...
> 
> <TouchableOpacity onPress={() => {}} />
> ```
> 
> This worked for me.

this is not a good solution. TouchableOpacity acts differently from react-native-gesture-handler. flex: 1 acting different

**Comment by gperler** [comment]
Any updates on this? The bug is still there. onPressIn worked for me in the development build on the device but not in the productive build. Also using newArchitecture with expo 52.

**Comment by serdarovez** [comment]
in my case the problem was the back button if i use image  in back button it broke my header. i removed it and everything works fine

**Comment by HannahCarney** [comment]
I used Touchable from gesture-handler and it works

**Comment by alberalgo** [comment]
onPressOut instead of onPress is working for me, thanks!

**Comment by nonamez** [comment]
Have same problem using react native button - it suddenly stopped to respond. Any luck for workaround without additional packages?

**Comment by KMJ-007** [comment]
facing the same issue

**Comment by IvoTsochev** [comment]
> Hei there As a workaround, use onPressOut instead of onPress Will work

Had the issue on some Android phones, "onPressOut" seems to be working properly.

**Comment by ice-cap0** [comment]
`onPressOut` will fire even when outside pressable area, which means users can't cancel the press by dragging out

**Comment by Thenlie** [comment]
Just wanted to add my experience here as I am seeing the same problem.

- Android only
- Release builds only
- Physical device only

I was not able to upgrade react-native-screens (due to a local config issue) so I went with the onPressOut work around. This is not ideal, but it does work for me.

```diff
--- a/src/routes/devToolStack.js
+++ b/src/routes/devToolStack.js
@@ -1,10 +1,11 @@
 import React from 'react';
 import createStackNavigator from '@/compLib/components/navigation/createStackNavigator';
 import { NavigationActions } from '../managers/NavigationManager';
-import { Button } from 'react-native';
+import { TouchableOpacity, Text } from 'react-native';
 import { navigatorRoutes, routeConfig } from './routes';
 import { lazyLoadComponent, lazyLoadWithRetry } from './suspense';
 import CustomModals from './event/CustomModals';
 
 const AccessTokenDetailScreen = React.lazy(() =>
   lazyLoadWithRetry(() => import('@/screens/devTools/AccessTokenDetailScreen'))
@@ -128,12 +129,20 @@ const devNavigatorRoutes = navigatorRoutes.DevToolsNavigator.screens;
 const navOptionsHome = screenName => ({
   options: {
     title: screenName,
     headerLeft: () => (
-      <Button
-        onPress={NavigationActions.exitDevMenu}
-        title="< Back"
+      <TouchableOpacity
+        style={{
+          backgroundColor: '#2196F3',
+          padding: 4
+        }}
+        /**
+         * onPressOut is used to work around the below issue with react-navigation
+         * https://github.com/react-navigation/react-navigation/issues/12274
+         */
+        onPressOut={NavigationActions.exitDevMenu}
         testID={'developer-tools-back'}
-      />
+      ><Text>{'< Back'}</Text></TouchableOpacity>
     ),
   },
 });
```

Thanks everyone for the suggestions here. When I am able to upgrade react-native-screens I will try to remember to edit my comment

**Comment by Issa-KODE** [comment]
> Try to use TouchableOpacity of react-native-gesture-handler
> 
> ```
> import { TouchableOpacity } from "react-native-gesture-handler";
> 
> ...
> 
> <TouchableOpacity onPress={() => {}} />
> ```
> 
> This worked for me.

It's good ! Thanks you

**Comment by htutwaiphyoe** [comment]
To fix this issue, we need to use TouchableOpacity.

**Comment by AlexCernik** [comment]
Any solution? :/

**Comment by jamalx31** [comment]
Only solution worked for us so far is using `onPressOut`. why this is not fixed yet?
